### PR TITLE
Create product Page

### PR DESCRIPTION
**Pull request by jayesh-parmar**

### CODE REVIEW

File: app/Http/Controllers/ProductController.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+use App\Models\Color;
+use App\Models\Product;
+use App\Models\Size;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    public function index()
+    {
+        $products = Product::with(['brand','color','size'])->paginate(10);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: load required columns while using relation

File: app/Http/Controllers/ProductController.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    
+        return view('admin.pages.product.index', ['products' => $products]);
+    }
+    public function add()
+    {
+        $brands=Brand::get();
+        $colors = Color::get();
+        $sizes = Size::get();
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: proper indentation & select required columns. 
do not forget select required columns

File: app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return view('admin.pages.product.add',['brands'=>$brands,'colors'=>$colors,'sizes'=>$sizes]);
+    }
+    public function store(Request $request)
+    {
+        $request->validate([
+            'name' => 'required|unique:products,name|max:255',
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]:
```suggestion
            'name' => ['required', 'unique:products,name', 'name', 'max:255']
```

always use array sytax

Comment (rajan-parmar) [reply app/Http/Controllers/ProductController.php]: also add other request field in validation

File: app/Http/Controllers/ProductController.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        ]);
+
+        return redirect()->route('product.index')->with('success',  'Product Added successfully ');
+    }
+    public function edit(string $productId)
+    {
+        $brands = Brand::get();
+        $colors = Color::get();
+        $sizes = Size::get();
+        $products = Product::find($productId);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]:
```suggestion
        $product = Product::find($productId);
```

File: app/Http/Controllers/ProductController.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        
+        return view('admin.pages.product.update', ['product' => $products, 'brands' => $brands, 'colors' => $colors, 'sizes' => $sizes]);
+    }
+
+    public function update(Request $request, string $productId)
+    {
+        $request->validate([
+            'name' => 'required|unique:brands,name,'. $productId .'|max:255',
+        ]);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: use laravel formrequest class for better code & maintanable.

File: app/Models/Brand.php [hunk 6]
```diff
@@ -9,7 +9,13 @@ class Brand extends Model
 {
     use HasFactory,HasUuids;
 
+   protected $primaryKey = 'id';
```
Comment (rajan-parmar) [inline app/Models/Brand.php]: not required

File: app/Models/Brand.php [hunk 7]
```diff
@@ -9,7 +9,13 @@ class Brand extends Model
...
     protected $fillable = [
         'name',
     ];
+
+    public function product()
+    {
+        return $this->hasMany(Product::class, 'id');
+    }
```
Comment (rajan-parmar) [inline app/Models/Brand.php]: is required?

File: app/Models/Color.php [hunk 8]
```diff
@@ -14,4 +14,8 @@ class Color extends Model
     protected $fillable = [
         'name',
     ];
+    public function product()
+    {
+        return $this->hasMany(Product::class, 'id');
+    }
```
Comment (rajan-parmar) [inline app/Models/Color.php]: is required?

File: app/Models/Size.php [hunk 9]
```diff
@@ -12,4 +12,8 @@ class Size extends Model
     protected $fillable = [
         'name',
     ];
+    public function product()
+    {
+        return $this->hasMany(Product::class, 'id');
+    }
```
Comment (rajan-parmar) [inline app/Models/Size.php]: is requierd?

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 10]
```diff
@@ -18,4 +18,8 @@ public function up(): void
             $table->timestamps();
         });
     }
+    public function down(): void
+    {
+        Schema::dropIfExists('users');
+    }
```
Comment (rajan-parmar) [inline database/migrations/2014_10_12_000000_create_users_table.php]: not required

File: resources/views/admin/pages/product/add.blade.php [hunk 11]
```diff
@@ -0,0 +1,77 @@
+@extends('admin.layouts.main')
+@section('content')
+    <h4 class="mb-4 text-lg font-semibold text-gray-600 dark:text-gray-300">
+        Add New Color
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/add.blade.php]:
```suggestion
        Add Product
```

File: resources/views/admin/pages/product/add.blade.php [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    <div class=" text-white ">
+        {{ session('success') }}
+    </div>
+    @endif
+    <form action="{{route('product.store')}}" method="post" class="mt-5">
+        @csrf
+        <div class="px-4 py-3 mb-8 bg-white rounded-lg shadow-md dark:bg-gray-800">
+            <label class="block text-sm">
+                <span class="text-gray-700 dark:text-gray-400">Name</span>
+                <x-input placeholder="Enter Brand Name" name="name" type="text" value="" />
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/add.blade.php]: use html required attribute which compulsory input value

File: resources/views/admin/pages/product/index.blade.php [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+            <button class="px-5 py-2 bt-0 text-sm font-medium leading-5 text-white transition-colors duration-150 bg-purple-600 border border-transparent rounded-lg active:bg-purple-600 hover:bg-purple-700 focus:outline-none focus:shadow-outline-purple">
+                Add New Product
+            </button>
+        </a>
+    </div>
+    @if(session('success'))
+    <div class=" text-white ">
+        {{ session('success') }}
+    </div>
+    @endif
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/index.blade.php]: move to layout file. why repeat the code in each file.

File: routes/web.php [hunk 14]
```diff
@@ -52,4 +53,15 @@
...
+    Route::get('products', 'index')->name('index');
+    Route::get('products/create', 'add')->name('create');
+    Route::post('product', 'store')->name('store');
+    Route::get('products/{productId}/edit', 'edit')->name('edit');
+    Route::post('products/{productId}/update', 'update')->name('update');
+    
+});
+
+
+
```
Comment (rajan-parmar) [inline routes/web.php]: remove extra space/lines

File: resources/views/admin/pages/product/update.blade.php [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+                <span class="text-gray-700 dark:text-gray-400">
+                    Status
+                </span>
+                <div class="mt-2">
+                    <label class="inline-flex items-center text-gray-600 dark:text-gray-400">
+                        <input type="radio" class="text-purple-600 form-radio focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="status" value="0" />
+                        <span class="ml-2">Pending</span>
+                    </label>
+                    <label class="inline-flex items-center ml-6 text-gray-600 dark:text-gray-400">
+                        <input type="radio" class="text-purple-600 form-radio focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="status" value="1" />
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/update.blade.php]: make common add/update file.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/ProductController.php [hunk 16]
```diff
@@ -12,23 +13,24 @@ class ProductController extends Controller
 {
     public function index()
     {
-        $products = Product::with(['brand','color','size'])->paginate(10);
-    
-        return view('admin.pages.product.index', ['products' => $products]);
+        $products = Product::with([
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: relation ma columns lidhi to product ni query ma select columns ni jarur nathi?

File: app/Http/Controllers/ProductController.php [hunk 17]
```diff
@@ -12,23 +13,24 @@ class ProductController extends Controller
...
-        $colors = Color::get();
-        $sizes = Size::get();
+        $brands = Brand::select('id', 'name')->get();
+        $colors = Color::select('id', 'name')->get();
+        $sizes = Size::select('id', 'name')->get();
+
         return view('admin.pages.product.add',['brands'=>$brands,'colors'=>$colors,'sizes'=>$sizes]);
     }
-    public function store(Request $request)
+    public function store( ProductValidation $request )
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: space na ave

File: app/Http/Requests/ProductValidation.php [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        return true;
+    }
+    public function rules(): array
+    {
+        return [
+            'name' => ['required', 'unique:products,name,'.$this->productId. 'max:255'],
+            'brand' => ['required', 'max:255'],
+            'color' => ['required', 'max:255'],
+            'size' => ['required', 'max:255'],
```
Comment (rajan-parmar) [inline app/Http/Requests/ProductValidation.php]: check which key are you receiving in request.
accordingly the value use the laravel validation.

File: app/Http/Requests/ProductValidation.php [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+class ProductValidation extends FormRequest
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+    public function rules(): array
+    {
+        return [
+            'name' => ['required', 'unique:products,name,'.$this->productId. 'max:255'],
```
Comment (rajan-parmar) [inline app/Http/Requests/ProductValidation.php]: `$this->productId` any purpose?

File: app/Http/Controllers/ProductController.php [hunk 20]
```diff
@@ -12,23 +13,24 @@ class ProductController extends Controller
...
     {
-        $request->validate([
-            'name' => 'required|unique:products,name|max:255',
-        ]);
-
         Product::create([
             'name' => $request->name,
             'brand_id' => $request->brand,
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: wrong $request key name

File: app/Http/Controllers/ProductController.php [hunk 21]
```diff
@@ -41,20 +43,16 @@ public function store(Request $request)
...
     public function edit(string $productId)
     {
-        $brands = Brand::get();
-        $colors = Color::get();
-        $sizes = Size::get();
-        $products = Product::find($productId);
+        $brands = Brand::select('id', 'name')->get();
+        $colors = Color::select('id', 'name')->get();
+        $sizes = Size::select('id', 'name')->get();
+        $product = Product::find($productId);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: select the columns

File: app/Http/Controllers/ProductController.php [hunk 22]
```diff
@@ -41,20 +43,16 @@ public function store(Request $request)
...
-    public function update(Request $request, string $productId)
+    public function update(ProductValidation $request, string $productId)
     {
-        $request->validate([
-            'name' => 'required|unique:brands,name,'. $productId .'|max:255',
-        ]);
-
         $product = Product::find($productId);
         $product->name = $request->name;
         $product->brand_id = $request->brand;
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: use can use direct `->update()` method

File: app/Models/Product.php [hunk 23]
```diff
@@ -31,4 +30,4 @@ public function size()
     {
         return $this->belongsTo(Size::class, 'size_id');
     }
-}
```
Comment (rajan-parmar) [inline app/Models/Product.php]: no need to define `size_id`, `color_id`, `brand_id`

File: resources/views/admin/pages/product/add.blade.php [hunk 24]
```diff
@@ -1,76 +1,16 @@
...
-                        </label>
-                    </div>
-                </div>
-                <div class="mt-4">
-                    <button class="px-4 py-2 bt-3 text-sm font-medium leading-5 text-white transition-colors duration-150 bg-purple-600 border border-transparent rounded-lg active:bg-purple-600 hover:bg-purple-700 focus:outline-none focus:shadow-outline-purple">
-                        Save
-                    </button>
+            @include('admin.pages.product.product_add_update')
+            <div class="mt-4">
+                <button class="px-4 py-2 bt-3 text-sm font-medium leading-5 text-white transition-colors duration-150 bg-purple-600 border border-transparent rounded-lg active:bg-purple-600 hover:bg-purple-700 focus:outline-none focus:shadow-outline-purple">
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/add.blade.php]: make new form.blade.php file and combine the add & update blade code into form.blade file.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

**Comment by rajan-parmar** [comment]
@jayesh-parmar still many changes are pending & you give me for review. 
whenever you got pr changes check one by one change request and update the code accordingly. if you have doubt, let me know. to clear it.

File: app/Http/Controllers/ProductController.php [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+use App\Models\Size;
+class ProductController extends Controller
+{
+    public function index()
+    {
+        $products = Product::with([
+            'brand',   
+            'color',   
+            'size',    
+        ])->select('id','name','brand_id','color_id','size_id','status')->paginate(10);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: ideally select should be first then with

like `->select()->with()`

File: app/Http/Controllers/ProductController.php [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $product = Product::find($productId);
+        
+        $product->update([
+            'name' => $request->name,
+            'brand_id' => $request->brand_id,
+            'color_id' => $request->color_id,
+            'size_id' => $request->size_id,
+            'status' => $request->status,
+        ]);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: https://laravel.com/docs/10.x/validation#working-with-validated-input

File: app/Http/Requests/ProductValidation.php [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+            'brand_id' => ['required', 'max:255'],
+            'color_id' => ['required', 'max:255'],
+            'size_id' => ['required', 'max:255'],
+            'status' => ['required', 'max:255'],
+        ];
+        if ($this->isMethod('PUT')) {
+           $rules = [
+                'name' => 'required','unique:products,name,' . $this->productId . 'max:255'
+           ];    
+        }
```
Comment (rajan-parmar) [inline app/Http/Requests/ProductValidation.php]: why this?

File: app/Http/Controllers/ProductController.php [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+            'color_id' =>$request->color_id,
+            'size_id'=>$request->size_id,
+            'status'=>$request->status
+        ]);
+    
+        return redirect()->route('product.index')->with('success',  'Product Added successfully ');
+    }
+    public function edit(string $productId)
+    {
+        $product = Product::find($productId);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: brands,colors & sizes not required?

File: resources/views/admin/pages/product/form.blade.php [hunk 29]
```diff
@@ -0,0 +1,83 @@
+@extends('admin.layouts.main')
+@section('content')
+<h4 class="mb-4 text-lg font-semibold text-gray-600 dark:text-gray-300">
+    @if (isset($product)) Update @else Add @endif Product
+</h4>
+@if (isset($product))
+<form action="{{ route('product.update', $product->id) }}" method="post" class="mt-5">
+    @method('PUT')
+    @else
+    <form action="{{ route('product.store') }}" method="post" class="mt-5">
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/form.blade.php]: why use multiple `<form` ?

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: app/Http/Requests/ProductValidation.php (hunk 27 above)
Comment (jayesh-parmar) [reply app/Http/Requests/ProductValidation.php]: update vakhte same name update thay te mate

File: resources/views/admin/pages/product/form.blade.php (hunk 29 above)
Comment (jayesh-parmar) [reply resources/views/admin/pages/product/form.blade.php]: update and add mate banaviyu che

File: app/Http/Requests/ProductValidation.php [hunk 30]
```diff
@@ -0,0 +1,33 @@
...
+        ];
+        if ($this->isMethod('PUT')) {
+           $rules = [
+                'name' => 'required','unique:products,name,' . $this->productId . 'max:255',
+                'brand_id' => ['required', 'max:255'],
+                'color_id' => ['required', 'max:255'],
+                'size_id' => ['required', 'max:255'],
+                'status' => ['required', 'max:255'],
+           ];    
+        }
```
Comment (rajan-parmar) [inline app/Http/Requests/ProductValidation.php]: https://laravel.com/docs/10.x/validation#rule-unique

`Forcing A Unique Rule To Ignore A Given ID:`

File: app/Http/Controllers/ProductController.php [hunk 31]
```diff
@@ -0,0 +1,46 @@
...
+use App\Models\Color;
+use App\Models\Product;
+class ProductController extends Controller
+{
+    public function index()
+    {
+        $products = Product::select('id', 'name', 'brand_id', 'color_id', 'size_id', 'status')->with([
+            'brand',   
+            'color',   
+            'size',
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]: load required relation columns always...

File: app/Http/Controllers/ProductController.php [hunk 32]
```diff
@@ -0,0 +1,46 @@
...
+    {
+        $product = Product::find($productId);
+
+        return view('admin.pages.product.form', compact('product'));
+    }
+
+    public function update(ProductValidation $request, Product $product)
+    {
+        $products = $request->validated();
+        $product->update($products);
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php]:
```suggestion
        $product->update($request->validated());
```

File: app/Http/Requests/ProductValidation.php (hunk 27 above)
Comment (rajan-parmar) [reply app/Http/Requests/ProductValidation.php]: this is not proper way

File: resources/views/admin/pages/product/form.blade.php (hunk 29 above)
Comment (rajan-parmar) [reply resources/views/admin/pages/product/form.blade.php]: turnaroy operator no use karo

**Review by rajan-parmar — CHANGES_REQUESTED** [review]
still many previous request changes are pending and you give me for review. check all then give me for review.

File: app/Http/Controllers/ProductController.php (hunk 31 above)
Comment (rajan-parmar) [reply app/Http/Controllers/ProductController.php]: relation sathe required columns pan mention karvi

**Comment by rajan-parmar** [comment]
@jayesh-parmar again may changes are pending. I just explained over the call/anydesk. now make changes and give your final code for review.

File: app/Http/Controllers/ProductController.php (hunk 21 above)
Comment (rajan-parmar) [reply app/Http/Controllers/ProductController.php]: haju pan product ni query ma select columns ni jarur nathi?

File: app/Providers/AppServiceProvider.php [hunk 33]
```diff
@@ -2,6 +2,9 @@
 
 namespace App\Providers;
 
+use App\Models\Brand;
+use App\Models\Color;
+use App\Models\Size;
```
Comment (rajan-parmar) [inline app/Providers/AppServiceProvider.php]: revert back to default state.

File: resources/views/admin/pages/product/form.blade.php [hunk 34]
```diff
@@ -0,0 +1,78 @@
...
+                <span class="text-gray-700 dark:text-gray-400">Name</span>
+                <x-input placeholder="Enter Product Name" name="name" type="text" value="{{ old('name', $product->name ?? '') }}" required />
+                @if($errors->has('name'))
+                <span class="text-xs text-red-600 dark:text-red-400">{{ $errors->first('name') }}</span>
+                @endif
+            </label>
+            <label class="block mt-4 text-sm">
+                <span class="text-gray-700 dark:text-gray-400">Select Brand</span>
+                <select class="block w-full mt-1 text-sm dark:text-gray-300 dark:border-gray-600 dark:bg-gray-700 form-select focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="brand_id">
+                    <option value="" disabled selected>Select a Brand</option>
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/form.blade.php]: add `required` attribute which have required validation rule.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/ProductController.php [hunk 35]
```diff
@@ -0,0 +1,48 @@
...
+    public function add()
+    {
+        $brands = Brand::select('id', 'name')->get();
+        $colors = Color::select('id', 'name')->get();
+        $sizes = Size::select('id', 'name')->get();    
+        return view('admin.pages.product.form', compact('brands','colors','sizes'));
+    }
+    public function store(ProductValidation $request)
+    {
+        Product::create($request->validated());
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php:28]:
```suggestion
        Product::create($request->all());
```

File: app/Http/Controllers/ProductController.php [hunk 36]
```diff
@@ -0,0 +1,48 @@
...
+        $colors = Color::select('id', 'name')->get();
+        $sizes = Size::select('id', 'name')->get(); 
+        $product = Product::select('id','name','brand_id','size_id','color_id','status')->find($productId);
+
+        return view('admin.pages.product.form', compact('product','brands','colors','sizes'));
+    }
+
+    public function update(ProductValidation $request, Product $product)
+    {
+        $product->update($request->validated());
```
Comment (rajan-parmar) [inline app/Http/Controllers/ProductController.php:44]:
```suggestion
        $product->update($request->all());
```

File: app/Providers/AppServiceProvider.php [hunk 37]
```diff
@@ -6,19 +6,12 @@
...
     }
-
-    /**
-     * Bootstrap any application services.
-     */
     public function boot(): void
     {
-        //
+        
     }
```
Comment (rajan-parmar) [inline app/Providers/AppServiceProvider.php]: revert back to default state

File: database/migrations/2023_09_12_162426_create_products_table.php [hunk 38]
```diff
@@ -0,0 +1,22 @@
...
+{
+
+    public function up(): void
+    {
+        Schema::create('products', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+            $table->string('name')->unique();
+            $table->foreignUuid('brand_id')->nullable()->constrained();
+            $table->foreignUuid('color_id')->nullable()->constrained();
+            $table->foreignUuid('size_id')->nullable()->constrained();
```
Comment (rajan-parmar) [inline database/migrations/2023_09_12_162426_create_products_table.php]: brand_id is compulsory
color_id & size_id is nullable then set validation rule & html required accordingly.

File: database/migrations/2023_09_12_162426_create_products_table.php [hunk 39]
```diff
@@ -0,0 +1,22 @@
...
+            $table->boolean('status')->nullable();
```
Comment (rajan-parmar) [inline database/migrations/2023_09_12_162426_create_products_table.php]:
```suggestion
            $table->boolean('status')->default(1)->comment(1:active, 0: archived);
```

File: resources/views/admin/layouts/main.blade.php [hunk 40]
```diff
@@ -44,7 +43,7 @@
                                             @csrf
                                             <li class="flex">
                                                 <a class="inline-flex items-center w-full px-2 py-1 text-sm font-semibold transition-colors duration-150 rounded-md hover:bg-gray-100 hover:text-gray-800 dark:hover:bg-gray-800 dark:hover:text-gray-200" :href="route('logout')" onclick="event.preventDefault();
-                                                this.closest('form').submit();">
+                                                    this.closest('form').submit();">
```
Comment (rajan-parmar) [inline resources/views/admin/layouts/main.blade.php]: why use again route('logout') in `a` attribute?
use button

File: resources/views/admin/pages/product/form.blade.php [hunk 41]
```diff
@@ -0,0 +1,78 @@
...
+            <label class="block text-sm">
+                <span class="text-gray-700 dark:text-gray-400">Name</span>
+                <x-input placeholder="Enter Product Name" name="name" type="text" value="{{ old('name', $product->name ?? '') }}" required />
+                @if($errors->has('name'))
+                <span class="text-xs text-red-600 dark:text-red-400">{{ $errors->first('name') }}</span>
+                @endif
+            </label>
+            <label class="block mt-4 text-sm">
+                <span class="text-gray-700 dark:text-gray-400">Select Brand</span>
+                <select class="block w-full mt-1 text-sm dark:text-gray-300 dark:border-gray-600 dark:bg-gray-700 form-select focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="brand_id" required>
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/form.blade.php:18]: `required` always set based on table column & validation rule is required or not.

File: resources/views/admin/pages/product/index.blade.php [hunk 42]
```diff
@@ -0,0 +1,79 @@
...
+                        <td class="px-4 py-3 text-sm">
+                            {{ $product->color->name }}
+                        </td>
+                        <td class="px-4 py-3 text-sm">
+                            {{ $product->size->name }}
+                        </td>
+                        <td class="px-4 py-3 text-xs">
+                            @php
+                            if($product->status == 1)
+                            echo '<span class="px-2 py-1 font-semibold leading-tight text-green-700 bg-green-100 rounded-full dark:bg-green-700 dark:text-green-100">delivered</span>';
```
Comment (rajan-parmar) [inline resources/views/admin/pages/product/index.blade.php]: echo is not needed

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

File: app/Providers/AppServiceProvider.php [hunk 43]
```diff
@@ -6,10 +6,20 @@
...
+     */
     public function register(): void
     {
-        
+        // Default registration code here
     }
+    /**
+     * Bootstrap any application services.
+     *
+     * @return void
```
Comment (rajan-parmar) [inline app/Providers/AppServiceProvider.php]: je pela hatu te kari nakho ne. main branch ma tamne mali jashe.

File: resources/views/admin/layouts/main.blade.php (hunk 40 above)
Comment (rajan-parmar) [reply resources/views/admin/layouts/main.blade.php]: I said use button not `a` attribute.

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

**Review by rajan-parmar — APPROVED** [review]

File: app/Http/Controllers/BrandController.php [hunk 44]
```diff
@@ -20,21 +20,21 @@ public function addBrand()
 
     public function store(BrandRequest $request)
     {
-        Brand::create($request->validated());
+        Brand::create($request->all());
```
Comment (gauravmak) [inline app/Http/Controllers/BrandController.php]: Why this change?

File: app/Http/Controllers/ProductController.php [hunk 45]
```diff
@@ -0,0 +1,48 @@
...
+        $product = Product::select('id', 'name', 'brand_id', 'size_id', 'color_id', 'status')->find($productId);
+
+        return view('admin.pages.product.form', compact('product', 'brands', 'colors', 'sizes'));
+    }
+
+    public function update(ProductValidation $request, Product $product)
+    {
+        $product->update($request->all());
+
+        return redirect()->route('product.index')->with('success', 'Product Updated successfully.');
```
Comment (gauravmak) [inline app/Http/Controllers/ProductController.php]:
```suggestion
        return redirect()->route('product.index')->with('success', 'Product updated successfully.');
```

File: app/Http/Controllers/ProductController.php [hunk 46]
```diff
@@ -0,0 +1,48 @@
...
+        $brands = Brand::select('id', 'name')->get();
+        $colors = Color::select('id', 'name')->get();
+        $sizes = Size::select('id', 'name')->get();    
+        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes'));
+    }
+    public function store(ProductValidation $request)
+    {
+        Product::create($request->all());
+
+        return redirect()->route('product.index')->with('success',  'Product Added successfully ');
```
Comment (gauravmak) [inline app/Http/Controllers/ProductController.php]:
```suggestion
        return redirect()->route('product.index')->with('success',  'Product added successfully ');
```

File: app/Http/Requests/ProductValidation.php [hunk 47]
```diff
@@ -0,0 +1,24 @@
...
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class ProductValidation extends FormRequest
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
```
Comment (gauravmak) [inline app/Http/Requests/ProductValidation.php]: Not needed

File: resources/views/admin/pages/product/form.blade.php [hunk 48]
```diff
@@ -0,0 +1,72 @@
...
+                <div class="mt-2">
+                    <label class="inline-flex items-center text-gray-600 dark:text-gray-400">
+                        <input type="radio" checked @if(isset($product)) {{ $product->status == '0' ? 'checked' : '' }} @endif class="text-purple-600 form-radio focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="status" value="0" />
+                        <span class="ml-2">Pending</span>
+                    </label>
+                    <label class="inline-flex items-center ml-6 text-gray-600 dark:text-gray-400">
+                        <input type="radio" @if(isset($product)) {{ $product->status == '1' ? 'checked' : '' }} @endif class="text-purple-600 form-radio focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="status" value="1" />
+                        <span class="ml-2">delivered</span>
+                    </label>
+                </div>
```
Comment (gauravmak) [inline resources/views/admin/pages/product/form.blade.php]: What kind of status are these? And why static?

File: resources/views/admin/pages/product/form.blade.php [hunk 49]
```diff
@@ -0,0 +1,72 @@
...
+                @if($errors->has('name'))
+                <span class="text-xs text-red-600 dark:text-red-400">{{ $errors->first('name') }}</span>
+                @endif
+            </label>
+            <label class="block mt-4 text-sm">
+                <span class="text-gray-700 dark:text-gray-400">Select Brand</span>
+                <select class="block w-full mt-1 text-sm dark:text-gray-300 dark:border-gray-600 dark:bg-gray-700 form-select focus:border-purple-400 focus:outline-none focus:shadow-outline-purple dark:focus:shadow-outline-gray" name="brand_id" required>
+                    <option value="" disabled selected>Select a Brand</option>
+                    @foreach ($brands as $brand)
+                    <option value="{{ $brand->id }}" @if(isset($product)) {{ $product->brand_id == $brand->id ? 'selected' : '' }} @endif>{{ $brand->name }}</option>
```
Comment (gauravmak) [inline resources/views/admin/pages/product/form.blade.php]: No need to use `old()` helper to pre-fill previously submitted value?

File: resources/views/admin/pages/product/index.blade.php [hunk 50]
```diff
@@ -0,0 +1,78 @@
...
+                        <th class="px-4 py-3">action</th>
+                    </tr>
+                </thead>
+                <tbody class="bg-white divide-y dark:divide-gray-700 dark:bg-gray-800">
+                    @foreach($products as $product)
+                    <tr class="text-gray-700 dark:text-gray-400">
+                        <td class="px-4 py-3 ">
+                            <div class="flex items-center text-sm">
+                                <div>
+                                    <p class="font-semibold">{{ $product->id }}</p>
```
Comment (gauravmak) [inline resources/views/admin/pages/product/index.blade.php]: No need to show ID

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/BrandController.php (hunk 44 above)
Comment (jayesh-parmar) [reply app/Http/Controllers/BrandController.php]: rajansir a kidhu thu  [https://github.com/jayesh-parmar/inventory-management/pull/16#discussion_r1329569230](url)

File: resources/views/admin/pages/product/form.blade.php (hunk 48 above)
Comment (jayesh-parmar) [reply resources/views/admin/pages/product/form.blade.php]: sir te add samye default panding  value store karva rakhyu thu

Comment (gauravmak) [reply resources/views/admin/pages/product/form.blade.php]: @rajan-parmar Please comment on this

File: app/Http/Controllers/BrandController.php (hunk 44 above)
Comment (gauravmak) [reply app/Http/Controllers/BrandController.php]: @rajan-parmar Please recreate the main issue as discussed

File: resources/views/admin/pages/product/form.blade.php (hunk 48 above)
Comment (rajan-parmar) [reply resources/views/admin/pages/product/form.blade.php]: @gauravmak I go with simple things in this PR as it's going long and after merged then I'll go for ENUM which [link](https://www.notion.so/Schedule-260b5b6bae41428d90a5576b8c200f97?p=d7fb20d331c7447d8257b9b6f84269c3&pm=s)

File: app/Http/Controllers/BrandController.php (hunk 44 above)
Comment (rajan-parmar) [reply app/Http/Controllers/BrandController.php]: @gauravmak I generate the issue at my side & mistake was `->ignore($this->product, 'id')` it should be `->ignore($this->product)`. same I explained to @jayesh-parmar over the call and he will make changes

**Review by rajan-parmar — APPROVED** [review]

File: app/Http/Requests/ProductValidation.php [hunk 51]
```diff
@@ -0,0 +1,20 @@
...
+
+class ProductValidation extends FormRequest
+{
+    public function rules(): array
+    {
+        return [
+            'name' => ['required', 'max:255', Rule::unique('products')->ignore($this->product)],
+            'brand_id' => ['required', 'string', 'max:255'],
+            'color_id' => ['nullable', 'string', 'max:255'],
+            'size_id' => ['string', 'nullable', 'max:255'],
```
Comment (gauravmak) [inline app/Http/Requests/ProductValidation.php]: Where is the Exists check? max rule not needed

**Review by gauravmak — CHANGES_REQUESTED** [review]

File: app/Http/Requests/ProductValidation.php (hunk 51 above)
Comment (gauravmak) [reply app/Http/Requests/ProductValidation.php]: @jayesh-parmar Please read the full like again.

You may contact @rajan-parmar if not sure.

**Comment by gauravmak** [comment]
@rajan-parmar Please check

File: app/Http/Requests/ProductValidation.php [hunk 52]
```diff
@@ -11,10 +11,10 @@ public function rules(): array
     {
         return [
             'name' => ['required', 'max:255', Rule::unique('products')->ignore($this->product)],
-            'brand_id' => ['required', 'string'],
-            'color_id' => ['nullable', 'string'],
-            'size_id' => ['string', 'nullable'],
+            'brand_id' => ['required', 'string', 'exists:brands,id'],
+            'color_id' => ['nullable', 'string', 'exists:colors,id'],
+            'size_id' => ['string', 'nullable', 'exists:sizes,id'],
```
Comment (rajan-parmar) [inline app/Http/Requests/ProductValidation.php]: rules follow the sequence.
here color_id & size_id are nullable in the table. so the rule should be nullable, then if get value then it will be a string/integer, and then it must exist...
so always follow the sequence

**Review by rajan-parmar — CHANGES_REQUESTED** [review]

**Review by rajan-parmar — APPROVED** [review]

**Comment by gauravmak** [comment]
@jayesh-parmar Please resolve conflicts
